### ISSUE TITLE
Immer enableMapSet stopped working with 1.5.1

### ISSUE THREAD

**Issue by rvanmil** [reduxjs/redux-toolkit#974]
Hi,

First off, I'm aware that using non-serializable values is not recommended 😊 That said, we prefer working with Map and Set in our state because they have a nicer api than array mutations. And so far this has been working just fine by using the `enableMapSet` plugin.

However this has now stopped working after updating from 1.5.0 to 1.5.1.
I have created a sandbox where the issue can be reproduced:

https://codesandbox.io/s/redux-toolkit-immer-issue-k3t1k

The sandbox will run and immediately show the error:
```
[Immer] The plugin for 'MapSet' has not been loaded into Immer. To enable the plugin, import and call `enableMapSet()` when initializing your application.
```

When switching the @reduxjs/toolkit dependency back to 1.5.0 the sandbox will render without errors.

I have also checked whether there are multiple versions of Immer installed (react-scripts also installs a version) and made sure to pin them all to 8.0.1 so they are all deduped and resolve to the same version, but nothing seems to work. Which is why I'm opening this issue because it now looks like redux toolkit 1.5.1 is the source of this problem.

**Comment by markerikson** [comment]
Hmm. That seems extremely odd and unlikely.  Nothing in 1.5.1 should have had any relation to this:

https://github.com/reduxjs/redux-toolkit/compare/v1.5.0..v1.5.1

The diff is noisy because of all the docs changes, but per https://github.com/reduxjs/redux-toolkit/releases/tag/v1.5.1 , you can see the actual code changes were very minimal.

I don't plan on spending any time investigating this myself, but if someone wants to look at it, go ahead.

**Comment by rvanmil** [comment]
I was afraid you were going say something like that 😉 I'll try to spend some more time investigating this.

**Comment by markerikson** [comment]
Yeah, I have no idea why there'd be any difference in behavior here, and it's not something that we specifically support or care about ourselves. So, given the other priorities I need to work on, I can't spend any time investigating this myself.

**Comment by rvanmil** [comment]
Well I've got a workaround, but I don't really understand what is happening and why it did work before. It looks like `createReducer` calls are executing before `enableMapSet` is called. I've got the enableMapSet call in my app's index.tsx entry. The reducer files all export the results of the createReducer calls. So I guess it seems logical that the createReducer calls are executed first, since all those files and dependencies are being imported before getting to the enableMapSet call.

Moving the entire contents of my index.tsx file into an app.tsx file and then replacing the index.ts with this fixes the problem:
```
import { enableMapSet } from 'immer'

enableMapSet()

import('./app').then(({ initialize }) => initialize())
```
The idea here was to delay all imports until after the call to enableMapSet. Seems to work 😊

I don't have a deep enough understanding of how webpack works and in which order imported stuff is executed to investigate any further. I also do not understand why things did work correctly before. At least I've got a workaround. Next step: remove Map and Set from the reducers and stick to serializable values as is recommended 😬
